### DISCUSSION TITLE
Talk on Commodore Users Europe

### DISCUSSION THREAD

**Discussion by intoinside** [c128lib/.github#4] [Team Posts]
Good morning to all!
I was invited by Sebastian Eggermont, a member of the Commodore Users Europe group (https://www.facebook.com/CommodoreUsersEurope), to present the library in a talk on March 4th.
I am very excited about this and I will have to work a bit to prepare the talk, add some missing pieces to the library, prepare some slides etc...
I'll keep you updated here on the progress, I hope the talk is confirmed because it would be very important as an advertisement for c128lib!
See you soon!

**Comment by johnpalermo** [comment]
This is great news. Looking forward to the updates.

**Comment by johnpalermo** [comment]
This is great news. Looking forward to your updates.  Also, hoping to contribute more here very soon.

**Comment by johnpalermo** [comment]
I saw that c128lib was featured on the website.  Will check the video soon!  😊

**Comment by intoinside** [comment]
Yes, they told me that video of separated talk will be also available on
youtube soon on Commodore users Europe channel, probably in these days. It
was an interesting night, i saw very interesting projects!

Il sab 11 mar 2023, 01:18 John Palermo ***@***.***> ha
scritto:

**Reply by johnpalermo** [reply]
I've been very busy with life, as it always happens, but I saw the presentation today and it was excellent.  It inspired me to contribute more.  Hopefully, I can do so.  It's taking me too long to finish the string library, but I'm almost done.  I like testing a lot and maybe sometimes too much.   Anyway, I enjoyed the presentation and was very informative.

**Comment by intoinside** [comment]
Hi John! Don't worry, I too haven't been able to work much on the library,
I'm trying to specialize the operation on the Vdc chip but unfortunately
there is little documentation available and I waste a lot of time
researching and testing the routines.
I saw that you generated a project to test the framework, could it make
sense to include it among the c128lib projects? How do you use it for
testing? I'm interested!

Il giorno lun 3 apr 2023 alle ore 00:01 John Palermo <
***@***.***> ha scritto:

> I've been very busy with life, as it always happens, but I saw the
> presentation today and it was excellent. It inspired me to contribute more.
> Hopefully, I can do so. It's taking me too long to finish the string
> library, but I'm almost done. I like testing a lot and maybe sometimes too
> much. Anyway, I enjoyed the presentation and was very informative.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/c128lib/discussions/4#discussioncomment-5504163>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB6EJ7BWGQFDCUIQQL3PLKDW7HZKNANCNFSM6AAAAAAVEKP2OQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
about.me/intoinside

**Reply by johnpalermo** [reply]
Ciao Raffaele,  

Sure, always takes time to research. The chipset library is already quite useful. 

The repository I created for testing the framework was intended as a backup of my test workspace in VSCode, as I often like to use GitHub to backup my local code. 

The structure I created was just from my head. I created 3 separate asm files just to manage the test sets better. A test asm file just contains various test cases for each macro. I define my preconditions and post conditions for each macro call and then set a breakpoint before and after. I check the post conditions inside of the debugger. For StringLength, for example, I check that Y is set to the length of the string. I pass the test case if the post conditions are met. I log the result in a CSV file in the same folder. 

I was thinking of also using these as the base for future documentation and samples, because they show quite well how to use the macros and their features, like passing strings by absolute or indirect addresses. 

I'm sure that there are better testing methodologies, but for me, this was quicker than learning something new. 

I think it's a good idea to incorporate something like this into the c128 projects. Let me know your thoughts on how it should be done. 

I have to learn more about CircleCI. Can that be an option?
